### PR TITLE
feat(autofix): Switch root cause step to give relevant code context for issue instead of suggested fixes

### PR DESCRIPTION
**Pull request by roaga**
Issue #947 and #970 

This reduces redundancy between the root cause and planning steps and helps the user understand and choose a root cause more confidently. This is by removing "suggested fixes" and replacing them with code snippets and other information to give the user context into the root cause.

This PR also includes some prompt tweaks to help it make sense with the LLM agents too (indirectly fixing #988 too), and some XML changes to maintain reliability with the new code context highlighting.

These changes resulted in an error weighted score of 0.45 and a regular score of 0.59, compared to 0.44 and 0.53 previously (an 11% improvement in quality and a 2% improvement in reliability; our best scores recorded so far!). 

This PR requires this frontend PR as well: https://github.com/getsentry/sentry/pull/75294

Screenshot 2024-07-31 at 12 43 50 PM

### CODE REVIEW

File: src/seer/automation/autofix/models.py [hunk 1]
```diff
@@ -76,12 +76,12 @@ class CustomRootCauseSelection(BaseModel):
     custom_root_cause: str
 
 
-class SuggestedFixRootCauseSelection(BaseModel):
+class CodeContextRootCauseSelection(BaseModel):
     cause_id: int
     fix_id: int
```
Comment (jennmueng) [inline src/seer/automation/autofix/models.py:81]: Why don't we make the selection be the entire root cause now and include all the snippets? We don't need to select a specific fix/snippet anymore right

File: src/seer/configuration.py [hunk 2]
```diff
@@ -77,6 +77,14 @@ def do_validation(self):
...
             assert self.has_sentry_integration, "Sentry integration required for production mode."
             assert self.SENTRY_DSN, "SENTRY_DSN required for production!"
+            assert self.GITHUB_APP_ID, "GITHUB_APP_ID required for production!"
+            assert self.GITHUB_PRIVATE_KEY, "GITHUB_PRIVATE_KEY required for production!"
+            assert (
+                self.DEV or self.GITHUB_SENTRY_APP_ID
+            ), "GITHUB_SENTRY_APP_ID required for production!"
+            assert (
+                self.DEV or self.GITHUB_SENTRY_PRIVATE_KEY
+            ), "GITHUB_SENTRY_PRIVATE_KEY required for production!"
```
Comment (jennmueng) [inline src/seer/configuration.py:87]: Should rebase as this has changed FYI

File: src/seer/automation/autofix/components/root_cause/models.py [hunk 3]
```diff
@@ -9,44 +9,41 @@
...
+class SnippetPromptXml(PromptXmlModel, tag="code"):
     file_path: str = attr()
     snippet: Annotated[str, StringConstraints(strip_whitespace=True)]
 
     @classmethod
     def get_example(cls):
         return cls(
             file_path="path/to/file.py",
-            snippet="# This snippet is optional. If there is a direct code fix then include it, if not then don't\ndef foo():\n    return 'bar'\n",
+            snippet="def foo():\n    return 'bar'\n",
```
Comment (jennmueng) [inline src/seer/automation/autofix/components/root_cause/models.py:20]: Can we ask the LLM to highlight a specific line in order to do: https://github.com/getsentry/sentry/pull/75294#discussion_r1698830602?

File: src/seer/configuration.py (hunk 2 above)
Comment (jennmueng) [reply src/seer/configuration.py:87]: looks like it's duplicating now lol

**Review by jennmueng — APPROVED** [review]
LGTM but fix the rebasing issue before merging

File: src/seer/automation/autofix/models.py (hunk 1 above)
Comment (roaga) [reply src/seer/automation/autofix/models.py]: Done

**Pull request merged**